group_project_v2/stage.py

```diff
@@ -0,0 +1,624 @@
+from collections import OrderedDict
+from datetime import datetime
+import json
+import logging
+
+from lazy.lazy import lazy
+import pytz
+import webob
+
+from xblock.core import XBlock
+from xblock.fields import Scope, String, DateTime
+from xblock.fragment import Fragment
+from xblock.validation import ValidationMessage
+
+from xblockutils.studio_editable import StudioEditableXBlockMixin, StudioContainerXBlockMixin
+
+from group_project_v2.api_error import ApiError
+from group_project_v2.mixins import ChildrenNavigationXBlockMixin, UserAwareXBlockMixin, CourseAwareXBlockMixin, \
+    WorkgroupAwareXBlockMixin, XBlockWithComponentsMixin
+from group_project_v2.stage_components import (
+    PeerSelectorXBlock, GroupSelectorXBlock,
+    GroupProjectReviewQuestionXBlock, GroupProjectPeerAssessmentXBlock, GroupProjectGroupAssessmentXBlock,
+    GroupProjectResourceXBlock, GroupProjectSubmissionXBlock,
+    StageState
+)
+from group_project_v2.project_api import project_api
+from group_project_v2.utils import loader, format_date, gettext as _, make_key
+
+log = logging.getLogger(__name__)
+
+
+class StageType(object):
+    NORMAL = 'normal'
+    UPLOAD = 'upload'
+    PEER_REVIEW = 'peer_review'
+    PEER_ASSESSMENT = 'peer_assessment'
+    GROUP_REVIEW = 'group_review'
+    GROUP_ASSESSMENT = 'group_assessment'
+
+
+class ResourceType(object):
+    NORMAL = 'normal'
+    OOYALA_VIDEO = 'ooyala'
+
+
+class BaseGroupActivityStage(
+    XBlock, XBlockWithComponentsMixin, StudioEditableXBlockMixin, StudioContainerXBlockMixin,
+    CourseAwareXBlockMixin, ChildrenNavigationXBlockMixin, UserAwareXBlockMixin
+):
+    submissions_stage = False
+
+    display_name = String(
+        display_name=_(u"Display Name"),
+        help=_(U"This is a name of the stage"),
+        scope=Scope.content,
+        default="Group Project V2 Stage"
+    )
+
+    open_date = DateTime(
+        display_name=_(u"Open Date"),
+        help=_(u"Stage open date"),
+        scope=Scope.settings
+    )
```

Comment:
@antoviaque it supposed to show a datepicker when focused. Probably installation issue as it is handled by xblock-utils. I'll get back to fresh install as soon as I finish what I'm doing right now.